server/src/main/scala/za/co/absa/atum/server/api/provider/PostgresAccessProvider.scala

```diff
@@ -19,28 +19,85 @@ package za.co.absa.atum.server.api.provider
 import com.typesafe.config.{Config, ConfigValueFactory}
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
-import za.co.absa.atum.server.api.AtumConfig
+import za.co.absa.atum.server.api.{AtumConfig, RetrieveAwsSecret}
 import za.co.absa.atum.server.api.database.Runs
 import za.co.absa.fadb.slick.FaDbPostgresProfile.api._
 import za.co.absa.fadb.slick.SlickPgEngine
-
 import scala.concurrent.ExecutionContext
 
+import scala.util.Try
+
 @Component
 class PostgresAccessProvider @Autowired()(atumConfig: AtumConfig) {
 
   val executionContext: ExecutionContext = ExecutionContext.Implicits.global
 
+  private val awsProfile = atumConfig.awsProfile
+  val retrieveAwsSecret: RetrieveAwsSecret = new RetrieveAwsSecret(awsProfile)
+
+  private val secretsSection = "fromSecrets"
+
+  private val connectionPool = "connectionPool"
+  private val dataSourceClass = "dataSourceClass"
+  private val serverName = "properties.serverName"
+  private val portNumber = "properties.portNumber"
+  private val databaseName = "properties.databaseName"
+  private val user = "properties.user"
+  private val password = "properties.password"
+
+  private def getKeysForSecrets(secretsConfig: Config): Map[String, String] = {
+    import PostgresAccessProvider._
+
+    Map
+      .empty[String, String]
+      .addSecretName(secretsConfig, connectionPool)
+      .addSecretName(secretsConfig, dataSourceClass)
+      .addSecretName(secretsConfig, serverName)
+      .addSecretName(secretsConfig, portNumber)
+      .addSecretName(secretsConfig, databaseName)
+      .addSecretName(secretsConfig, user)
+      .addSecretName(secretsConfig, password)
+  }
+
+  private def overrideWithSecret(oldConfig: Config, path: String, secretName: String): Config = {
+
+    val secretString: Seq[String] = retrieveAwsSecret.retrieveAwsSecret(secretName)
+    val overrideValue = secretString.foldLeft(Try("")) {(acc, s) =>
```

Comment:
What's the reason for using Try here? What are you trying to accomplish here, please explain.

Reply:
So that it can catch the exception in case it happens during the of string concatenation operation.